### ISSUE TITLE
LSP not starting when opening new project files

### ISSUE THREAD

**Issue by martin-dimi** [doomemacs/doomemacs#7369]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27 or 28. *(Doom does not support development builds like 29+ or any version ending in .50 or .9x)*

### Expected behavior

* Open workspace `X` file `file1.tsx`
* LSP starts/works when in `file1.tsx`
* Move to another file (Go To Definition or normal open another file)  `file2.tsx`
* LSP starts/works when in `file2.tsx`

### Current behavior

* Open workspace `X` file `file1.tsx`
* LSP starts/works when in `file1.tsx`
* Move to another file (Go To Definition or normal open another file)  `file2.tsx`
* LSP DOES NOT starts/works when in `file2.tsx`

### Steps to reproduce

1. Open doom emacs
2. Switch to project and open any tsx file
3. Check if LSP starts (it should)
4. Move to another file (either go to definition or just open another file) from the same project.
5. LSP does not work/start in that file

I noticed `lsp-describe-session` does not include `file2.tsx` upon opening it. If I run `M-x lsp` then `file2.tsx` is added to the session and LSP works as expected for that file.

### System Information

https://pastebin.com/0LG33Yeu

**Comment by martin-dimi** [comment]
Oh, I'm running on Emacs 29. Will try on 28 and update...

**Comment by rollge** [comment]
checkout to commit 24601b300e87e909340524476713dcaaa5d095cd is fine to me.

**Comment by rollge** [comment]
> checkout to commit [24601b3](https://github.com/doomemacs/doomemacs/commit/24601b300e87e909340524476713dcaaa5d095cd) is fine to me.

i think it's a bug but i don't know why, so i just roll back to 24601b300e87e909340524476713dcaaa5d095cd

**Comment by martin-dimi** [comment]
Hmm, not sure if I'm doing it correct, but I checked out to `833be11` and ran `doom build`. Unfortunately, I'm still seeing the issue

**Comment by Shooooooooo** [comment]
Does your emacs have tree-sitter? Someone in Emacs China forum was seeing similar issue.
[tree sitter 导致lsp失效](https://emacs-china.org/t/tree-sitter-lsp/25141)

**Comment by martin-dimi** [comment]
@Shooooooooo I am running tree sitter. Not sure what to take away from that forum. A bit hard to follow

**Comment by Shooooooooo** [comment]
Check what is the major-mode for your file. If it’s ***-ts-mode instead of ***-mode, it won’t be recognized by lsp. That thread was describing a workaround which I have not tried.
